auto_dev/protocols/formatter.py

```diff
@@ -29,8 +29,21 @@
 from auto_dev.protocols.primitives import PRIMITIVE_TYPE_MAP
 
 
-def render_field(field: Field) -> str:
-    field_type = PRIMITIVE_TYPE_MAP.get(field.type, field.type)
+def qualified_type(adapter: FileAdapter | MessageAdapter, type_name: str) -> str:
+
+    def find_definition(scope):
+        if scope is None or isinstance(scope, FileAdapter):
+            return None
+        if type_name in scope.enum_names or type_name in scope.message_names:
+            return f"{scope.fully_qualified_name}.{type_name}"
+        return find_definition(scope.parent)
+
+    qualified_name = find_definition(adapter)
+    return qualified_name if qualified_name is not None else PRIMITIVE_TYPE_MAP.get(type_name, type_name)
```

Comment:
helper function to make life easy in the recursion when trying to scaffold the correctly (fully) qualified type